javascript/src/index.ts

```diff
@@ -10,6 +11,7 @@ import RegularExpression from './RegularExpression.js'
 import { Expression } from './types.js'
 
 export {
+  AbstractCompiler,
```

Comment:
Maybe we could also export `RegExpCompiler`? That may be useful sometime?

Reply:
It's only used internally, so I'd rather not. By exposing as little as possible we minimise the API surface to maintain.